pysr/ssr.py

```diff
@@ -0,0 +1,171 @@
+            https://symbolicml.org/DynamicQuantities.jl/dev/units/ for more
+            information.
+            Length should be equal to n_features.
+
+        Returns
+        -------
+        self : object
+            Fitted estimator.
+        """
+
+        if y is not None:
+            warnings.warn(
+                "Recursive symbolic regression does not use `y` - this parameter is being ignored"
+            )
+        if y_units is not None:
+            warnings.warn(
+                "Recursive symbolic regression does not use `y_units` - this parameter is being ignored"
+            )
+        if Xresampled is not None:
+            warnings.warn(
+                "Recursive symbolic regression does not use `Xresampled` - this parameter is being ignored"
+            )
+
+        if self.recursive_history_length <= 0:
+            raise ValueError(
+                "The `recursive_history_length` parameter must be greater than 0 (otherwise it's not recursion)."
+            )
+        if len(X.shape) > 2:
+            raise ValueError(
+                "Recursive symbolic regression only supports up to 2D data; please flatten your data first"
+            )
+        elif len(X) < 2:
+            raise ValueError(
+                "Recursive symbolic regression requires at least 2 datapoints; if you tried to pass a 1D array, use array.reshape(-1, 1)"
+            )
+        if len(X) <= self.recursive_history_length + 1:
+            raise ValueError(
+                f"Recursive symbolic regression with a history length of {self.recursive_history_length} requires at least {self.recursive_history_length + 2} datapoints."
+            )
+        if isinstance(weights, np.ndarray) and len(weights) != len(X):
+            raise ValueError("The length of `weights` must have shape (n_times,).")
+        y = X.copy()
+        temp = X.copy()[0]
+        X = np.lib.stride_tricks.sliding_window_view(
+            y[:-1].flatten(), self.recursive_history_length * temp.shape[0]
+        )[:: temp.shape[0], :]
+        y = np.array([i for i in y[self.recursive_history_length :]])
+        y_units = X_units
+        if isinstance(weights, np.ndarray):
+            weights = weights[self.recursive_history_length :]
+
+        if not variable_names:
+            if len(temp.shape) == 0:
+                variable_names = [
+                    f"xt_{i}" for i in range(self.recursive_history_length, 0, -1)
+                ]
+            elif len(temp.shape) == 1:
+                variable_names = [
+                    f"x{i}t_{j}"
+                    for j in range(temp.shape[0])
+                    for i in range(self.recursive_history_length, 0, -1)
+                ]
+        else:
+            variable_names = [
+                i + "t_" + str(j)
+                for i in variable_names
+                for j in range(self.recursive_history_length, 0, -1)
+            ]
```

Comment:
Could this be split out to a separate function? Then it can be unittested without needing to call the entire `.fit` code.